src/paho/mqtt/client.py

```diff
@@ -1042,16 +1214,19 @@ def reconnect(self):
                 if not self.suppress_exceptions:
                     raise
 
-        sock = self._create_socket_connection()
```

Comment:
The changes from here up to where `self._sock` gets assigned smell like this should be refactored to a separate `self._create_socket()` function (which would probably type more nicely).